src/mjolnir/ingest_transit.cc

```diff
@@ -257,18 +257,29 @@ void setup_stops(Transit& tile,
   auto onestop_id = isGenerated ? tile_stop.stop_id + "_" + to_string(node_type) : tile_stop.stop_id;
   node->set_onestop_id(onestop_id);
   if (node_type == NodeType::kMultiUseTransitPlatform) {
-    stop_graphIds[{onestop_id, feed_path}] = node_id;
+    // when this platform is not generated, we can use its actual given id verbatim because thats how
+    // other gtfs entities will refer to it. however when it is generated we must use its parents id
+    // and not the generated one because the references in the feed have no idea that we are doing
+    // the generation of new ideas for non-existant platforms
+    platform_node_ids[{tile_stop.stop_id, feed_path}] = node_id;
```

Comment:
the problem with giving these generated platforms a new id is that that id is not known to any other entities in the gtfs feed. any schedule information (trips) that connection stations together will connect them by the stations id since from the gtfs feeds point of view there is no platform, valhalla is making that up. so later, when we want to look up with graphid we gave to a platform that we made up by its id, ie when we are iterating over the gtfs feeds trips we need to remember those graphids by their parent station id not their new made up generated id, because of course the feeds raw data doesnt know about that, its a valhalla convention